opentelemetry-api/src/opentelemetry/attributes/__init__.py

```diff
@@ -118,6 +118,98 @@ def _clean_attribute(
     return None
 
 
+def _clean_extended_attribute_value(
+    value: types.AnyValue, max_len: Optional[int]
+) -> types.AnyValue:
+    # for primitive types just return the value and eventually shorten the string length
+    if value is None or isinstance(value, _VALID_ATTR_VALUE_TYPES):
+        if max_len is not None and isinstance(value, str):
+            value = value[:max_len]
+        return value
+
+    if isinstance(value, Mapping):
+        cleaned_dict: dict[str, types.AnyValue] = {}
+        for key, element in value.items():
+            # skip invalid keys
+            if not (key and isinstance(key, str)):
+                _logger.warning(
```

Comment:
nit but warning level might be a bit spammy. it's likely to be a recurring problem

Reply:
This matches validation of the non-extended attributes